emacs/merlin.el

```diff
@@ -1921,7 +1941,7 @@ Empty string defaults to jumping to all these."
          (with-temp-buffer
            (if (eq (call-process-shell-command
```

Comment:
Now both branches of this `if` return the same value: `"ocamlmerlin"` while before the opam bin dir was prepended to it in one case. 

I guess it was useless to return the absolute path to the executable since the `$PATH` variable is replaced by the path to the opam bin folder. Is that correct ? If it is, then it's a good thing to get rid of it, but we should switch that `if` to only perform the side effect of modifying the path, returning the same value in both branches is misleading.

Reply:
I guess it was useless to return the absolute path to the executable since the `$PATH` variable is replaced by the path to the opam bin folder. Is that correct ?

Yes!

> If it is, then it's a good thing to get rid of it, but we should switch that `if` to only perform the side effect of modifying the path, returning the same value in both branches is misleading.

Agree; I just updated the function accordingly; thanks!

Reply:
Thanks for the changes! I think we can go one step further and make the value much closer to the binding name:
```
unless command
  setq command "ocamlmerlin"
  with-temp-buffer-query-opam
```

Reply:
Hi @voodoos!

I don't think so… because there are other cases after the current `(setq command` :

```
       (cond
        ((functionp merlin-command) (funcall merlin-command))
        ((stringp merlin-command) merlin-command)
```